processing/src/main/java/org/apache/druid/segment/StringDimensionIndexer.java

```diff
@@ -60,20 +60,29 @@ private static String emptyToNullIfNeeded(@Nullable Object o)
   private final MultiValueHandling multiValueHandling;
   private final boolean hasBitmapIndexes;
   private final boolean hasSpatialIndexes;
+  private final boolean useMaxMemoryEstimates;
   private volatile boolean hasMultipleValues = false;
 
-  public StringDimensionIndexer(MultiValueHandling multiValueHandling, boolean hasBitmapIndexes, boolean hasSpatialIndexes)
+  public StringDimensionIndexer(
+      MultiValueHandling multiValueHandling,
+      boolean hasBitmapIndexes,
+      boolean hasSpatialIndexes,
+      boolean useMaxMemoryEstimates
+  )
   {
+    super(useMaxMemoryEstimates ? new DimensionDictionary<>() : new StringDimensionDictionary());
```

Comment:
nit: this seems strange/confusing, why wouldn't `StringDimensionIndexer` always use `StringDimensionDictionary` here? it seems like could just pass in a value of `false` to control the value of `computeOnHeapSize` instead of sometimes not using `StringDimensionDictionary`

Reply:
You are right, it is weird.
Fixed it.